pkg/cmd/roachtest/tests/roachtest.go

```diff
@@ -53,4 +54,62 @@ func registerRoachtest(r registry.Registry) {
 		Timeout: 3 * time.Minute,
 		Cluster: r.MakeClusterSpec(3),
 	})
+
+	// Manual test for verifying framework behavior in a test failure scenario
+	// via unexpected node fatal error from an explicitly monitored node
+	r.Add(registry.TestSpec{
+		Name:             "roachtest/manual/monitor/test-failure/node-fatal-explicit-monitor",
+		Owner:            registry.OwnerTestEng,
+		Cluster:          r.MakeClusterSpec(1),
+		CompatibleClouds: registry.AllClouds,
+		Suites:           registry.ManualOnly,
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			monitorFatalTest(ctx, t, c)
+		},
+	})
+
+	// Manual test for verifying framework behavior in a test failure scenario
+	// via unexpected node fatal error using the roachtest built-in node monitor
+	r.Add(registry.TestSpec{
+		Name:             "roachtest/manual/monitor/test-failure/node-fatal-global-monitor",
+		Owner:            registry.OwnerTestEng,
+		Cluster:          r.MakeClusterSpec(1),
+		CompatibleClouds: registry.AllClouds,
+		Suites:           registry.ManualOnly,
+		Monitor:          true,
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			monitorFatalTestGlobal(ctx, t, c)
+		},
+	})
+}
+
+// monitorFatalTest will always fail with a node logging a fatal error in a
+// goroutine that is being watched by a monitor
+func monitorFatalTest(ctx context.Context, t test.Test, c cluster.Cluster) {
+	c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings())
+	m := c.NewDeprecatedMonitor(ctx, c.Node(1))
+	n1Conn := c.Conn(ctx, t.L(), 1)
```

Comment:
Nit: since Go is a statically-typed language, the suffix is redundant; i.e., we know the type of `n1` is `*gosql.DB`.